snapshot/config_snapshot.py

```diff
@@ -0,0 +1,228 @@
+            ), "Diagnostics data is not a dictionary"
+            for key, value in diagnostics_data_names.items():
+                if value == "scalar":
+                    self._write_data_names[key] = False
+                elif value == "vector":
+                    self._write_data_names[key] = True
+                else:
+                    raise Exception(
+                        "Diagnostics data dictionary has a value other than 'scalar' or 'vector'"
+                    )
+        except BaseException:
+            self._logger.info(
+                "No diagnostics data is defined. Snapshot computation will not output any diagnostics data."
+            )
+
+        try:
+            if data["diagnostics"]["output_micro_sim_solve_time"]:
+                self._output_micro_sim_time = True
+                self._write_data_names["micro_sim_time"] = False
+        except BaseException:
+            self._logger.info(
+                "Snapshot Computation will not output time required to solve each micro simulation."
+            )
+
+    def get_parameter_file_name(self):
+        """
+        Get the name of the parameter file.
+
+        Returns
+        -------
+        parameter_file_name : string
+            Name of the hdf5 file containing the macro parameters.
+        """
+
+        return self._parameter_file_name
+
+    def get_read_data_names(self):
+        """
+        Get the user defined dictionary carrying information of the data to be read from preCICE.
+
+        Returns
+        -------
+        read_data_names: dict_like
+            A dictionary containing the names of the data to be read from preCICE as keys and information on whether
+            the data are scalar or vector as values.
+        """
+        return self._read_data_names
+
+    def get_write_data_names(self):
+        """
+        Get the user defined dictionary carrying information of the data to be written to preCICE.
+
+        Returns
+        -------
+        write_data_names: dict_like
+            A dictionary containing the names of the data to be written to preCICE as keys and information on whether
+            the data are scalar or vector as values.
+        """
+        return self._write_data_names
+
+    def get_micro_file_name(self):
+        """
+        Get the path to the Python script of the micro-simulation.
+
+        Returns
+        -------
+        micro_file_name : string
+            String carrying the path to the Python script of the micro-simulation.
+        """
+        return self._micro_file_name
+
+    def write_micro_solve_time(self):
+        """
+        Depending on user input, Snapshot computation will calculate execution time of solve() step of every micro simulation.
+
+        Returns
+        -------
+        output_micro_sim_time : bool
+            True if micro simulation solve time is required.
+        """
+        return self._output_micro_sim_time
+
+    def get_postprocessing(self):
```

Comment:
This is essentially a boolean, right? We should have a more intuitive name of the function.